src/server/orders/matcher.py

```diff
@@ -92,6 +92,16 @@ def add(self, order: Order) -> MatchResult:
         Match the order against the current order book, producing a new order book and a series of executions.
         The final execution price will be the new market price of the securities.
         """
+
+        if order.type == OrderType.market:
+            return self.match_market_order(order)
+        elif order.type == OrderType.limit:
+            return self.match_limit_order(order)
```

Comment:
Massive code duplication in these, but prioritising ease of debugging (low branching factor) over DRYness